org.compiere.db.postgresql.provider/src/org/compiere/db/DB_PostgreSQL.java

```diff
@@ -891,40 +680,117 @@ public Connection getDriverConnection (String dbUrl, String dbUid, String dbPwd)
 		getDriver();
 		return DriverManager.getConnection (dbUrl, dbUid, dbPwd);
 	}	//	getDriverConnection
+	
+	private Properties getPoolProperties() {	
+		//check property file from home
+		File userPropertyFile = new File(getPoolPropertiesFile());
+		URL propertyFileURL = null;
+		
+		if (userPropertyFile.exists() && userPropertyFile.canRead())
+		{			
+			try {
+				propertyFileURL = userPropertyFile.toURI().toURL();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+			
+		if (propertyFileURL == null)
+		{
+			propertyFileURL = PostgreSQLBundleActivator.bundleContext.getBundle().getEntry("META-INF/pool/server.default.properties");						
+		}
 
+		Properties poolProperties = new Properties();
+		try (InputStream propertyFileInputStream = propertyFileURL.openStream()) {
+			poolProperties.load(propertyFileInputStream);
+		} catch (Exception e) {
+			throw new DBException(e);
+		} 
 
+		//auto create property file at home folder from default config
+		if (!userPropertyFile.exists())			
+		{
+			try {				
+				Path directory = userPropertyFile.toPath().getParent();
+				Files.createDirectories(directory);
+								
+				try (InputStream propertyFileInputStream = propertyFileURL.openStream()) {
+					Files.copy(propertyFileInputStream, userPropertyFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+				} 
+			
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		
+		return poolProperties;
+	}
+	
+	/** Boolean to indicate the PostgreSQL connection pool is either initializing or initialized.*/
+	private final AtomicBoolean initialized = new AtomicBoolean(false);
+	/** Latch which can be used to wait for initialization completion. */
+	private final CountDownLatch initializedLatch = new CountDownLatch(1); 
+	
+	/**
+	 * Allows the connection pool to be lazily initialized. While it might be preferable to do
+	 * this once upon initialization of this class the current design of iDempiere makes this 
+	 * hard.
+	 * 
+	 * Calling this method will block until the pool is configured. This does NOT mean it will
+	 * block until a database connection has been setup.
+	 * 
+	 * @param connection
+	 */
+	private void ensureInitialized(CConnection connection) {
+		if (!initialized.compareAndSet(false, true)) {
+			try {
+				initializedLatch.await();
+			} catch (InterruptedException e) {
+				return;
+			}
+		} else {
+			Properties poolProperties = getPoolProperties();		
+			
+	        try
+	        {
+	            m_ds.setDriverClassName(DRIVER);
+	            
+	            for(String propertyKey : poolProperties.stringPropertyNames()) {
+	            	m_ds.addDataSourceProperty(propertyKey, poolProperties.get(propertyKey));

```

Comment:
this is not working, only cachePrepStmts, prepStmtCacheSize and prepStmtCacheSqlLimit is dataSourceProperty, the other are connection pool properties.

[db_postgresql.patch.txt](https://github.com/idempiere/idempiere/files/9275872/db_postgresql.patch.txt)

I use this patch in my local testing but perhaps there are better approach.

Also, note that due to the way logging is currently configure in iDempiere, if you use HikariConfig instead, any invalid property will freeze the system (stuck in the logger).

Reply:
I've changed the way the pool is initialized and the settings files are now loaded correctly.